pkg/api/handlers/compat/images.go

```diff
@@ -356,6 +356,14 @@ loop: // break out of for/select infinite loop
 					Message: msg,
 				}
 				report.ErrorMessage = msg
+				var errcd errcode.ErrorCoder
+				var statusCode int
+				if errors.As(err, &errcd) {
+					statusCode = errcd.ErrorCode().Descriptor().HTTPStatusCode
+				} else {
+					statusCode = http.StatusInternalServerError
+				}
+				w.WriteHeader(statusCode)
```

Comment:
It seems that this can happen after progress has been reported, in principle. How does this work? Looking at the `WriteHeader` implementation, I’d expect at least `superfluous response.WriteHeader call from …` warnings to show up somewhere.

Reply:
Consider writing a helper function to write the header exactly once.  Once it has been written, subsequent headers should be ignored by clients (or cause errors, I guess).

Reply:
Writing a header string in the middle of the JSON stream would not work at all (and the `WriteHeader` package does not actually _do that_ — but it does output a warning to some output stream, I haven’t checked where that ends up).

I do think it would be cleaner to only actually call `WriteHeader` once, at the right time — to avoid that warning, but more importantly to make it easy to understand for readers of the code what is _actually_ happening.